zombies.js

```diff
@@ -79,135 +94,226 @@
+      console.log(`${Player.name} placed ${item.name} into pack`);
+      this.getPack().push(item);
+      return true;
+    } else {
+      console.log(`Pack is too full`);
+      return false;
+    }
+  }
+
+  /**
+   * Player Class Method => discardItem(item)
+   * -----------------------------
+   * Player discards an item from their pack.
+   *
+   * Use Array's indexOf method to check if the pack contains the item.
+   * If an item is not found in the pack, indexOf returns -1.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf
+   *
+   * If the item is in the pack, remove it from the pack using Array's splice method.
+   * Print the player and item names and a message saying the item was discarded.
+   * Return true for the successful discard.
+   * Note: The splice method can also be used for array element replacement.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
+   *
+   * If the item is not in the pack, return a message with the item name saying
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+  discardItem(item) {
+    let itemIndex = this.getPack().indexOf(item);
+    if (itemIndex === -1) {
+      console.log(`${item} cannot be found`);
+      return false;
+    } else if (itemIndex >= 0) {
+      this.getPack().splice(itemIndex, 1);
+      console.log(`Removed ${item}`);
+      return true;
+    }
+  }
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  equip(itemToEquip) {
+    let itemIndex = this.getPack().indexOf(itemToEquip);
+    let isWeapon = itemToEquip instanceof Weapon;
+
+    if (isWeapon && itemIndex >= 0) {
+      if (this.equipped === false) {
+        this.getPack().splice(itemIndex, 1);
+        this.equipped = itemToEquip;
+      } else {
+        let currentWeapon = this.equipped;
+        this.getPack().push(currentWeapon);
+        this.getPack().splice(itemIndex, 1);
```

Comment:
The logic here should be reversed, you should remove the itemToEquip first, then add in the currently equipped item.